presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/RowExpressionFormatter.java

```diff
@@ -123,12 +144,30 @@ public String visitConstant(ConstantExpression node, ConnectorSession session)
             }
 
             Type type = node.getType();
-            if (node.getType().getJavaType() == Block.class) {
+            if (type.getJavaType() == Block.class) {
                 Block block = (Block) value;
                 // TODO: format block
                 return format("[Block: position count: %s; size: %s bytes]", block.getPositionCount(), block.getRetainedSizeInBytes());
             }
-            return type.getDisplayName().toUpperCase() + " " + value.toString();
+
+            String valueString = "'" + value.toString().replace("'", "''") + "'";
+
+            if (VarbinaryType.isVarbinaryType(type)) {
+                return "X" + valueString;
+            }
+
+            String typeString;
+            if (Chars.isCharType(type)) {
+                typeString = StandardTypes.CHAR.toUpperCase();
+            }
+            else if (Decimals.isShortDecimal(type) || Decimals.isLongDecimal(type)) {
+                typeString = StandardTypes.DECIMAL.toUpperCase();
+            }
+            else {
+                typeString = type.getDisplayName().toUpperCase();
+            }
+
+            return typeString + " " + valueString;
```

Comment:
No space between type and value

Reply:
It seems like a space is allowed:
```
presto:sf1> SELECT bigint'5', TYPEOF(bigint'5'), bigint '5', TYPEOF(bigint '5'); 
 _col0 | _col1  | _col2 | _col3  
-------+--------+-------+--------
     5 | bigint |     5 | bigint 
```
Is it better to have no space anyways?

Reply:
ya more like a nit.

Reply:
done